llvm/lib/Target/NVPTX/NVPTXISelDAGToDAG.cpp

```diff
@@ -1502,6 +1542,16 @@ bool NVPTXDAGToDAGISel::tryStoreVector(SDNode *N) {
     N2 = N->getOperand(5);
     ToTypeWidth = TotalWidth / 4;
     break;
+  case NVPTXISD::StoreV8:
+    if (!Subtarget->has256BitMaskedLoadStore())
+      return false;
+    VecType = NVPTX::PTXLdStInstCode::V8;
+    Ops.append({N->getOperand(1), N->getOperand(2), N->getOperand(3),
+                N->getOperand(4), N->getOperand(5), N->getOperand(6),
+                N->getOperand(7), N->getOperand(8)});
+    N2 = N->getOperand(9);
```

Comment:
Can we pull some of this logic out of the switch here as well. I think it would be cleanest if the switch only produced the number of elements (perhaps as a static helper function or lambda). And then we just used that number to derive all these other variables.

Reply:
We would still need to set the VecType in the switch, right? Or would you want a second switch that switches from NumElts -> VecType?

Reply:
I think we should be able to just remove VecType from these instructions all together. If not, it should be simply a mater of casting the number of elements to the VecType.